utilidades/storage_file_utils.py

```
@@ -679,94 +677,59 @@ def _filter_df_for_partition(self, df: pd.DataFrame, partition, partition_cols:
+            if final_df_sorted is None:
+                raise ValueError("Failed to prepare data for writing")
+
+            # Write file
+            row_group_size = min(self.row_group_size, len(final_df_sorted))
+            if row_group_size == 0:
+                raise ValueError("No rows to write")
+
+            self._write_final_parquet(final_df_sorted, output_file_path, value_cols, row_group_size)
+
+        except Exception as e:
+            raise Exception(f"Failed to combine and write partition: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)